File: src/main/java/net/vidageek/games/regex/task/TaskGroup.java
package net.vidageek.games.regex.task;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Iterator;
import java.util.List;

import net.vidageek.games.regex.Descriptions;
import net.vidageek.games.task.Task;

final public class TaskGroup implements Iterable<Task> {

	private final List<Task> tasks;
	private final Descriptions descriptions;
	private final String groupName;
	private final String name;

	public TaskGroup(String name, final String groupName, final Descriptions descriptions) {
		this.name = name;
		this.groupName = groupName;
		this.descriptions = descriptions;
		tasks = new ArrayList<Task>();
	}

	public void add(final Task task) {
		tasks.add(task);
	}

	public int size() {
		return tasks.size();
	}

	public Task task(final int i) {
		return tasks.get(i);
	}

	public Iterator<Task> iterator() {
		return Collections.unmodifiableCollection(tasks).iterator();
	}

	public String getDescription() {
		return descriptions.forGroup(groupName);
	}

	public String getName() {
		return name;
	}

}

File: src/main/scala/vggames/regex/task/TaskGroup.scala
package vggames.regex.task

import java.util.{Collections, ArrayList}

import scala.collection.JavaConversions.asScalaIterator

import vggames.shared.task.{Task, Descriptions}

class TaskGroup(name : String, groupName : String, descriptions : Descriptions) extends Iterable[Task] {
  val tasks = new ArrayList[Task]

  def add(task : Task) : Unit = tasks.add(task)

  override def size = tasks.size

  def task(index : Int) = tasks.get(index)

  def iterator = Collections.unmodifiableCollection(tasks).iterator

  def getDescription = descriptions.forGroup(groupName)

  def getName = name
}